### ISSUE TITLE
vulkan-tools 1.3.254 build issue against vulkan-headers 1.3.254 and vulkan-loader 1.3.254

### ISSUE THREAD

**Issue by chenrui333** [KhronosGroup/Vulkan-Tools#816]
👋 running into some issue of [building vulkan-tools 1.3.254](https://github.com/Homebrew/homebrew-core/pull/133987) against vulkan-headers 1.3.254 and vulkan-loader 1.3.254 (my molten-vk is 1.2.4 as shown in [known_good.json](https://github.com/KhronosGroup/Vulkan-Tools/blob/ca8bb4ee3cc9afdeca4b49c5ef758bad7cce2c72/scripts/known_good.json)). 

The error is as below:

```
In file included from /opt/homebrew/include/vulkan/vulkan.hpp:6814:
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18683:9: error: use of undeclared identifier 'structureChains'; did you mean 'StructureChain'?
        structureChains.resize( dataSize );
        ^
/opt/homebrew/include/vulkan/vulkan.hpp:983:9: note: 'StructureChain' declared here
  class StructureChain : public std::tuple<ChainElements...>
        ^
In file included from /tmp/vulkan-tools-20230617-54130-vujd57/Vulkan-Tools-1.3.254/cube/macOS/cubepp/DemoViewController.mm:24:
In file included from /tmp/vulkan-tools-20230617-54130-vujd57/Vulkan-Tools-1.3.254/cube/cube.cpp:43:
In file included from /opt/homebrew/include/vulkan/vulkan.hpp:6814:
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18683:9: error: use of undeclared identifier 'structureChains'; did you mean 'StructureChain'?
        structureChains.resize( dataSize );
        ^
/opt/homebrew/include/vulkan/vulkan.hpp:983:9: note: 'StructureChain' declared here
  class StructureChain : public std::tuple<ChainElements...>
        ^
In file included from /tmp/vulkan-tools-20230617-54130-vujd57/Vulkan-Tools-1.3.254/cube/macOS/cubepp/DemoViewController.mm:24:
In file included from /tmp/vulkan-tools-20230617-54130-vujd57/Vulkan-Tools-1.3.254/cube/cube.cpp:43:
In file included from /opt/homebrew/include/vulkan/vulkan.hpp:6814:
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18683:9: error: unknown type name 'structureChains'
        structureChains.resize( dataSize );
        ^
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18683:24: error: cannot use dot operator on a type
        structureChains.resize( dataSize );
                       ^
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18687:27: error: use of undeclared identifier 'structureChains'
          data[i].pNext = structureChains[i].template get<void>().pNext;
                          ^
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18687:46: error: 'template' keyword not permitted here
          data[i].pNext = structureChains[i].template get<void>().pNext;
                                             ^
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18732:9: error: use of undeclared identifier 'structureChains'; did you mean 'StructureChain'?
        structureChains.resize( dataSize );
        ^
/opt/homebrew/include/vulkan/vulkan.hpp:983:9: note: 'StructureChain' declared here
  class StructureChain : public std::tuple<ChainElements...>
        ^
In file included from /tmp/vulkan-tools-20230617-54130-vujd57/Vulkan-Tools-1.3.254/cube/macOS/cubepp/DemoViewController.mm:24:
In file included from /tmp/vulkan-tools-20230617-54130-vujd57/Vulkan-Tools-1.3.254/cube/cube.cpp:43:
In file included from /opt/homebrew/include/vulkan/vulkan.hpp:6814:
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18732:9: error: use of undeclared identifier 'structureChains'; did you mean 'StructureChain'?
        structureChains.resize( dataSize );
        ^
/opt/homebrew/include/vulkan/vulkan.hpp:983:9: note: 'StructureChain' declared here
  class StructureChain : public std::tuple<ChainElements...>
        ^
In file included from /tmp/vulkan-tools-20230617-54130-vujd57/Vulkan-Tools-1.3.254/cube/macOS/cubepp/DemoViewController.mm:24:
In file included from /tmp/vulkan-tools-20230617-54130-vujd57/Vulkan-Tools-1.3.254/cube/cube.cpp:43:
In file included from /opt/homebrew/include/vulkan/vulkan.hpp:6814:
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18732:9: error: unknown type name 'structureChains'
        structureChains.resize( dataSize );
        ^
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18732:24: error: cannot use dot operator on a type
        structureChains.resize( dataSize );
                       ^
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18736:27: error: use of undeclared identifier 'structureChains'
          data[i].pNext = structureChains[i].template get<void>().pNext;
                          ^
/opt/homebrew/include/vulkan/vulkan_funcs.hpp:18736:46: error: 'template' keyword not permitted here
          data[i].pNext = structureChains[i].template get<void>().pNext;
                                             ^
12 errors generated.
```

**Comment by charles-lunarg** [comment]
This is an issue with vulkan-hpp that only is present when `VK_ENABLE_BETA_EXTENSIONS` is set. This repo has worked around the issue by not enabling `VK_ENABLE_BETA_EXTENSIONS` on all targets, rather by only adding it to the projects that need it (like vulkaninfo & mock_icd, both of which do not include any vulkan-hpp headers).

**Comment by charles-lunarg** [comment]
This commit is the update to 1.3.254 and ostensibly removes `VK_ENABLE_BETA_EXTENSIONS` from everything. 
https://github.com/KhronosGroup/Vulkan-Tools/commit/98fe631044dfde2841b7b4204adb22b72627bd4f

So if you are referencing that commit, you shouldn't be getting that error (unless something *else* is enabling VK_ENABLE_BETA_EXTENSIONS...)


Side note, I looked at your PR and `-DTOOLS_CODEGEN=ON` only *enables* codegen, it does not run it. Nor should you need to run the codegen at all, as it is all checked in to the repo.

**Comment by chenrui333** [comment]
Thanks @charles-lunarg for the pointer, this got resolved in Homebrew/homebrew-core#136287
